cas-server-core/src/main/java/org/jasig/cas/web/view/AbstractCasView.java

```diff
@@ -211,7 +211,7 @@ protected final Service getServiceFrom(final Map<String, Object> model) {
 
         final Assertion assertion = getAssertionFrom(model);
         final List<Authentication> chainedAuthentications = assertion.getChainedAuthentications();
-        for (int i = 0; i < chainedAuthentications.size() - 2; i++) {
+        for (int i = 0; i < chainedAuthentications.size() - 1; i++) {
```

Comment:
Agreed, but this is implemented directly via the protocol. The very last index is always the primary authn, and everything else before it is the proxy chain that get pushed down the list in the response to note the order. I'll see if I can pull this out into a smaller more meaningful methof.